app/actions/websocket/users.test.ts

```diff
@@ -272,4 +292,267 @@ describe('WebSocket Users Actions', () => {
+                mockRejectedValue(new Error('test error'));
+
+            const msg = {
+                data: {
+                    user_id: 'user1',
+                    values: {field1: 'value1'},
+                },
+            } as WebSocketMessage;
+
+            await handleCustomProfileAttributesValuesUpdatedEvent(
+                serverUrl,
+                msg,
+            );
+
+            expect(logUtils.logError).toHaveBeenCalled();
+        });
+    });
+
+    describe('handleCustomProfileAttributesFieldUpdatedEvent', () => {
+        it('should handle missing operator', async () => {
+            jest.spyOn(logUtils, 'logError').mockImplementation(() => {});
+            DatabaseManager.serverDatabases = {};
+            const msg = {
+                data: {
+                    field: {id: 'field1'},
+                },
+            } as WebSocketMessage;
+
+            await handleCustomProfileAttributesFieldUpdatedEvent(
+                serverUrl,
+                msg,
+            );
+
+            expect(logUtils.logError).toHaveBeenCalled();
+        });
+
+        it('should handle custom profile field update', async () => {
+            const mockHandleCustomProfileFields = jest.
+                fn().
+                mockResolvedValue([]);
+            operator.handleCustomProfileFields = mockHandleCustomProfileFields;
+
+            const mockField = {
+                id: 'field1',
+                group_id: 'group1',
+                name: 'Field 1',
+                type: 'text',
+                target_id: 'target1',
+                target_type: 'user',
+                create_at: 1000,
+                update_at: 2000,
+                delete_at: 0,
+                attrs: {required: true},
+            };
+
+            const msg = {
+                data: {
+                    field: mockField,
+                },
+            } as WebSocketMessage;
+
+            await handleCustomProfileAttributesFieldUpdatedEvent(
+                serverUrl,
+                msg,
+            );
+
+            expect(mockHandleCustomProfileFields).toHaveBeenCalledWith({
+                fields: [mockField],
+                prepareRecordsOnly: false,
+            });
+        });
+
+        it('should handle errors during field update', async () => {
+            jest.spyOn(logUtils, 'logError').mockImplementation(() => {});
+            operator.handleCustomProfileFields = jest.
+                fn().
+                mockRejectedValue(new Error('test error'));
+
+            const msg = {
+                data: {
+                    field: {id: 'field1'},
+                },
+            } as WebSocketMessage;
+
+            await handleCustomProfileAttributesFieldUpdatedEvent(
+                serverUrl,
+                msg,
+            );
+
+            expect(logUtils.logError).toHaveBeenCalled();
+        });
+    });
+
+    describe('handleCustomProfileAttributesFieldDeletedEvent', () => {
```

Comment:
none of these tests actually checks for the fields and attributes to have been deleted from the db

Reply:
this is now happening at the handler level